### ISSUE TITLE
Would compatibility with PiSugar be possible?

### ISSUE THREAD

**Issue by LTCharlton** [bverc/retropie-status-overlay#22]
There is an I2C table in its documentation (towards the end of its GitHub) but I'm not sure how that works. If it's possible and not too hard what are some of the steps to implement it?

https://github.com/PiSugar/PiSugar/wiki/PiSugar-3-Series

**Comment by bverc** [comment]
This certainly would be possible, although without hardware makes it very difficult to test and verify. 

This is a similar issue to #17 . My plan was to add some sort of modular plugin system that allows connection of different battery sensing options, but this has been on the backburner.

You are welcome to fork and add the functionality you need, then I'll have a starting point to make into some sort of modular system.

**Comment by LTCharlton** [comment]
Thank you for your input. I'll fork it and take a look. Thank you.

**Comment by bverc** [comment]
Well I have added the modular adc support, and I have made a rough attempt at getting something together for the PiSugar. However, I cannot test it without hardware. Could you have a look at the pisugar-test branch, and see if you can get it going?

Check you have I2C enabled in raspi-config. 

In the config.ini, set `ADCType = PISUGAR`, and `ADCGain = 1`

**Comment by kineticmedic** [comment]
I'm new to github so I'm not entirely sure how to submit it, but I modified the pisugar.py script from the branch to correctly pull the voltage locally for me and would be happy to provide it.

**Comment by bverc** [comment]
Absolutely. The correct way would be to create your own fork from the pisugar-test branch, then create a pull request.

Alternatively, just post the code changes below, and I can create for you.

**Comment by kineticmedic** [comment]
I think I've done the pull request correctly. I may also take a stab at implementing charging status indicators (is charging) to the overlay as well, at least as far as the pi sugar since that's what I have access to... depends on how tricky and how much time I have.

**Comment by bverc** [comment]
Fixed in #40

**Comment by bverc** [comment]
Thanks @kineticmedic for your edits. Have merged your changes in and added you as author to that module.
